### ISSUE TITLE
Error message of failed tests are hard to parse.

### ISSUE THREAD

**Issue by rainwoodman** [bccp/nbodykit#189]
It is hard to see what command the pipeline has ran, making it hard to reproduce the failed test case or debug.

```
[yfeng1@waterfall nbodykit]$ py.test -x nbodykit 
======================================================================================= test session starts =======================================================================================
platform linux2 -- Python 2.7.11 -- py-1.4.30 -- pytest-2.7.3
rootdir: /home/yfeng1/source/nbodykit, inifile: setup.cfg
plugins: pipeline, cov
collected 93 items 

nbodykit/test/test_batch.py .F

============================================================================================ FAILURES =============================================================================================
____________________________________________________________________________________ TestStdin.test_exit_code _____________________________________________________________________________________

self = <nbodykit.test.test_batch.TestStdin testMethod=test_exit_code>

    def test_exit_code(self):
>       asserts.test_exit_code(self)

/home/yfeng1/source/nbodykit/nbodykit/test/test_batch.py:35: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <nbodykit.test.test_batch.TestStdin testMethod=test_exit_code>

    def test_exit_code(self):
        """
        Assert that the exit code is equal to 0
        """
>       assert self.run_fixture.exit_code == 0
E       AssertionError: assert 127 == 0
E        +  where 127 = RunClass1(run_id=RunClass1_78d72c5f-e0f8-4531-82d9-cb2b99f6b83c, ...).exit_code
E        +    where RunClass1(run_id=RunClass1_78d72c5f-e0f8-4531-82d9-cb2b99f6b83c, ...) = <nbodykit.test.test_batch.TestStdin testMethod=test_exit_code>.run_fixture

/home/yfeng1/source/nbodykit/nbodykit/test/asserts.py:25: AssertionError
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
=============================================================================== 1 failed, 1 passed in 0.91 seconds ================================================================================
```

**Comment by rainwoodman** [comment]
Somehow all tests are failing on my computer...

**Comment by nickhand** [comment]
This is the test that reads from stdin -- which if I remember correctly didn't work on your computer? 

I thought the test would just hang if it didn't work properly so I added a timeout, but it doest seem to be working quite right. I thought it would just fail and then continue with the other tests, which is what happens on my machine, so I'm not sure why it is marked as `Interrupted` here.

We can add more verbose flags, i.e., `-vvv` to the `py.test` command for more output. I'll look into redirecting the logging output to a file and printing it when the test fails -- I think this is possible.

**Comment by rainwoodman** [comment]
I asked it to skip this error. Most errors are due to paintbrush. I need to add a subsample data source -- do you recommended me do this of the master branch?

**Comment by nickhand** [comment]
The logging output gets sent to stderr. py.test does have a feature where it captures stdout during the tests  (I'm not sure about stderr) and will automatically output it for failed tests. I think it is controlled by some command line arguments -- I'll see if I can get it to automatically do what we want here

Feel free to add it to either the master or develop branch -- I would probably say develop but I don't think it really matters.

**Comment by rainwoodman** [comment]
Do you have a cure for the paintbrush error I am seeing off the develop branch? travis is happy but my box isn't..

**Comment by nickhand** [comment]
what exactly is the error you are seeing for paintbrush? you've updated to the latest set of dependencies?

**Comment by rainwoodman** [comment]
It was because pmesh was too old. I think this means I can close #181 ? I lost track on it.

**Comment by nickhand** [comment]
The issue about the Lanzcos window? I'm not sure what the origin of that
issue is

**Comment by rainwoodman** [comment]
What are the choices of 'paintbrush'? I noticed the lanczos.py file in
pmesh, and thought we had plan to expose these via pmesh to nbodykit, or do
something similar.

**Comment by nickhand** [comment]
Right now the paintbrush can be 'cic' or 'tsc', but it doesn't use
Lanzcos.py at all, so we still need to expose that if we want to be able to
use it
